x-pack/plugins/triggers_actions_ui/public/application/sections/alert_summary_widget/components/alert_summary_widget_compact.tsx

```diff
@@ -37,23 +37,12 @@ export interface AlertSummaryWidgetCompactProps {
 export const AlertSummaryWidgetCompact = ({
   activeAlertCount,
   activeAlerts,
-  chartProps: { theme, baseTheme },
+  chartProps: { themeOverrides } = {},
   recoveredAlertCount,
   timeRangeTitle,
   onClick,
 }: AlertSummaryWidgetCompactProps) => {
-  const chartTheme = [
-    ...(theme ? [theme] : []),
-    EUI_SPARKLINE_THEME_PARTIAL,
-    {
-      chartMargins: {
-        left: 10,
-        right: 10,
-        top: 10,
-        bottom: 10,
-      },
-    },
-  ];
+  const { baseTheme, sparklineTheme } = useChartThemes();
```

Comment:
Q: The reason for passing the `theme, baseTheme` via `chartProps` was to ensure that the user of this component is aware of the dependency that this component has on `charts` service.
With this new implementation, we are implicitly using this dependency via `useChartThemes`, so I wonder how we can make sure that this dependency is included in the plugin that uses `AlertSummaryWidget`?

Reply:
Ok good to know let me look at it again.

Reply:
Ok @maryam-saeidi, thanks again for this comment.

I dug into this deeper and discussed it with the core team.

It's strange how we export static components from the plugin runtime contract, but until this point I didn't really think about it.

So instead of requiring the user pass the required `ChartProps` I propose a solution similar to the one below...

https://github.com/elastic/kibana/blob/bd48254819967289c8c2ef169c8285f980d05020/x-pack/plugins/index_management/public/plugin.ts#L115-L150

...In this example they are also exporting a react component but encapsulating the necessary dependencies and passing them to the component. I've done a similar thing for the `AlertSummaryWidget` component in 78f9b39ce8c221170e2955e9d1951b31936aaa13.

With this change, the consuming plugin that is using the `getAlertSummaryWidget` component only has to pass any desired overrides to the `chartProps` and not the global `baseTheme`. This also does not require the consuming plugin to list the `charts` plugin as `requiredPlugin` unless they are using it directly.

Let me know how you feel about these changes. Thanks!

Reply:
No it will not show this type of error. I tested this on the `x-pack/plugins/observability_solution/observability` plugin by removing the `charts` plugin from it's `requiredPlugins` and stripping out all of the usages of `charts` in the plugin.

In this setup, since we are now passing in the dependencies from plugin contract, `charts` is always defined in the props of the `AlertSummaryWidget`.

However, in the same setup, if I were to not pass the `charts` plugin directly to the `getAlertSummaryWidget` but instead get `charts` via `useKibana().services.charts`, this would show the error like you mentioned above saying something like `cannot access 'theme' of undefined`, because the `charts` plugin is undefined. Then you could fix this by simply including the `charts` plugin in the `requiredPlugin` of the plugin that is using/rendering the `AlertSummaryWidget`.

But by passing `charts` as a dependency within `plugin.ts` you avoid the need to require the `charts` plugin and guarantee the `charts` dependency is defined, hence avoiding the error.

Reply:
> In this case, what would happen if we load this component in a plugin that does not have charts as a required plugin

A more direct answer to this question - Any plugin that requires the `triggers_actions_ui` plugin, can use the `getAlertSummaryWidget` method without that plugin needing to add the `charts` plugin.